src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -294,6 +294,16 @@ public final String getLine(int index) {
         return context.get().fileContents.getLine(index);
     }
 
+    /**
+     * Returns the line associated with the tree.
+     *
+     * @param line Unicode code points of required line
+     * @return the array of Unicode code points
+     */
+    public static final int[] getCodePoints(String line) {
```

Comment:
Please make method parameter `int index`, similar to `getLine()`. This method should replace usages of `getLine()` in all affected checks. You can use `getLine()` method in `getCodePoints()` method to get String representation of line, then return code points.

Reply:
Also, after thinking on this a bit, it seems to me that a more descriptive method name, like `getCodePointsFromLine(int index)` might make our intention to get a specific line of text, in code points, from the source file more clear.

Reply:
Also, after thinking on this a bit, it seems to me that a more descriptive method name, like `getCodePointsFromLine(int index)` might make our intention to get a specific line of text, in code points, from the source file more clear. @rnveach do you agree?

Reply:
Usage of `getLine` cannot be replaced in checks which use particular line to check for a character, for eg: `line.substring(0, columnNo).trim().endsWith("*/")`, `braceLine.charAt(brace.getColumnNo() + 1) != '}')`, `braceLine.length()` here and some more. I can create a new method but right now `getLine()` serves the purpose. So can I know why `getLine` should be completed removed from checks?

Or do you mean something else?

Reply:
@nmancus1 What about `getLineCodePoints`? Either one works fine for me.

Reply:
Sorry, thinking of one test case got me confused, Printing AST made it clear.

Reply:
Resolved